lib/rspec/support/differ.rb

```diff
@@ -72,6 +76,46 @@ def initialize(opts={})
       end
 
     private
+      def run(actual, expected)
+        diff = ""
+
+        no_procs_no_numbers = lambda {|var1, var2| no_procs?(var1, var2) && no_numbers?(var1, var2)}
+
+        if all_strings?(actual, expected)
+          diff = diff_as_string(coerce_to_string(actual), coerce_to_string(expected)) if any_multiline_strings?(actual, expected)
+        elsif (keys = all_keys_from_hash(expected)) && keys.any?
```

Comment:
What if we check no_procs_no_numbers here, and the other check goes inside?

Reply:
Nice, that makes it easier to read! Thanks! I've submitted a new commit with your suggestion.

Reply:
Something broke specs, I hope it’s not because of this cosmetic suggestion. Please feel free to revert if this is the case